saml-core/src/main/java/org/keycloak/saml/processing/core/parsers/saml/assertion/SAMLConditionsParser.java

```diff
@@ -64,6 +65,11 @@ protected void processSubElement(XMLEventReader xmlEventReader, ConditionsType t
                 target.addCondition(oneTimeUseCondition);
                 break;
 
+            case PROXY_RESTRICTION:
+                ProxyRestrictionType proxyRestriction = new ProxyRestrictionType();
```

Comment:
The XML stream needs to handle this element properly. You should introduce a new `SAMLProxyRestrictionParser` class (almost the same as [`SAMLAudienceRestrictionParser`](https://github.com/keycloak/keycloak/blob/master/saml-core/src/main/java/org/keycloak/saml/processing/core/parsers/saml/assertion/SAMLAudienceRestrictionParser.java)), and parse the `ProxyRestriction` tag instead of creating empty `ProxyRestrictionType`:
```java
ProxyRestrictionType proxyRestriction = SAMLAudienceRestrictionParser.getInstance().parse(xmlEventReader);
```

Note that `SAMLProxyRestrictionParser` should handle `ProxyRestriction` attributes, namely `Count` (see e.g. [here](https://github.com/keycloak/keycloak/blob/master/saml-core/src/main/java/org/keycloak/saml/processing/core/parsers/saml/metadata/SAMLIndexedEndpointTypeParser.java#L32) for sample of this task)

Reply:
Will do that, we just needed a fast fix, and I didn't payed attention to the attribute :-)